test/Lock.t.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.23;
+
+import {Test} from "forge-std/Test.sol";
+import {MockLock} from "./utils/MockLock.sol";
+
+contract LockTest is Test {
+    MockLock public mockLock;
+
+    address public addressThis;
+    address public externalLocker = makeAddr("externalLocker");
+
+    bytes constant LOCKER_IS_THIS = abi.encodeWithSelector(bytes4(keccak256("lockerIsThis()")));
+    bytes constant LOCKER_IS_ANYONE = abi.encodeWithSelector(bytes4(keccak256("lockerIsAnyone()")));
+
+    function setUp() public {
+        mockLock = new MockLock();
+        addressThis = address(mockLock);
+    }
+
+    function test_lockerIsThis_succeeds() public {
```

Comment:
assert that _getLocker() is address(0) at the end of all these calls?

Reply:
Hmm since we are tstoring locker we would probably have to do the assert a wrapper function around the actual call to Lock?

Reply:
Bc otherwise it's guaranteed to be cleared after each call